packages/transformer/src/IModelTransformer.ts

```diff
@@ -2789,8 +2839,14 @@ export class IModelTransformer extends IModelExportHandler {
         this.context.remapElement(sourceElementId, targetElementId);
       }
     );
-    if (this._csFileProps === undefined || this._csFileProps.length === 0)
-      return;
+    this.exporter.addCustomChanges();
+
+    if (this._csFileProps === undefined || this._csFileProps.length === 0) {
+      if (this.exporter.sourceDbChanges?.isEmpty) return;
```

Comment:
`this._sourceChangeDataState` might end up being set to  "has-changes" when `this.exporter.sourceDbChanges` is `undefined` due to this if: `if (this.exporter.sourceDbChanges?.isEmpty)`.

Reply:
blah good catch, now returning if undefined || empty